example/eblob_backend.c

```diff
@@ -86,15 +86,28 @@ static int eblob_read_params_compare(const void *p1, const void *p2)
 }
 
 /* Pre-callback that formats arguments and calls ictl->callback */
-static int blob_iterate_callback_common(struct eblob_disk_control *dc, void *data, void *priv, int no_meta) {
+static int blob_iterate_callback_common(struct eblob_disk_control *dc, int fd, uint64_t data_offset, void *priv, int no_meta) {
 	struct dnet_iterator_ctl *ictl = priv;
 	struct dnet_ext_list elist;
 	struct eblob_backend_config *c = ictl->iterate_private;
 	uint64_t size;
+	void *data, *saved_data;
 	int err;
 
 	assert(dc != NULL);
-	assert(data != NULL);
+
+	size = dc->disk_size;
+	saved_data = data = malloc(size);
```

Comment:
This is not going to work with large files.
Why wouldn't you use new fd checksumming API?